src/main/java/org/folio/auth/authtokenmodule/MainVerticle.java

```diff
@@ -265,50 +256,50 @@ private void handleDecodeEncryptedToken(RoutingContext ctx) {
   private void handleRefresh(RoutingContext ctx) {
     try {
       logger.debug("Token refresh request from {}", ctx.request().absoluteURI());
+  
       if (ctx.request().method() != HttpMethod.POST) {
         endText(ctx, 400, "Invalid method for this endpoint");
         return;
       }
+  
       String content = ctx.getBodyAsString();
       JsonObject requestJson;
+  
       try {
-        requestJson = parseJsonObject(content,
-          new String[]{"refreshToken"});
+        requestJson = parseJsonObject(content, new String[] { "refreshToken" });
       } catch (Exception e) {
-        endText(ctx, 400, "Unable to parse content: ", e);
+        endText(ctx, 400, "Unable to parse content of refresh token request: ", e);
         return;
       }
-      String token = requestJson.getString("refreshToken");
-      String tokenContent;
-      JsonObject tokenClaims;
-      try {
-        tokenContent = tokenCreator.decodeJWEToken(token);
-        tokenClaims = new JsonObject(tokenContent);
-      } catch (Exception e) {
-        String message = String.format("Unable to decode token %s: %s",
-          token, e.getLocalizedMessage());
-        logger.error(message);
-        endText(ctx, 400, "Invalid token format");
-        return;
-      }
-      String tenant = ctx.request().headers().get(XOkapiHeaders.TENANT);
-      //Go ahead and make the new request token
-      String newAuthToken = mintNewAuthToken(tenant, tokenClaims);
-      validateRefreshToken(tokenClaims, ctx).onComplete(res -> {
-        if (res.failed()) {
-          endText(ctx, 500, res.cause());
-          return;
-        }
-        if (!res.result()) {
-          endText(ctx, 401, "Invalid refresh token");
+  
+      String encryptedJWE = requestJson.getString("refreshToken");
+      Future<Token> tokenValidationResult = Token.validate(encryptedJWE, tokenCreator, ctx.request());
+
+      tokenValidationResult.onFailure(h -> {
+        var e = (TokenValidationException)h;
```

Comment:
There is no guarantee that all Exceptions returned by a Future are a TokenValidationException.
```
if (h instanceof TokenValidationException) {
   ...
} else {
   ...
}
```